closed/src/java.base/share/classes/openj9/internal/criu/CRIUSECProvider.java

```diff
@@ -39,26 +42,45 @@ public final class CRIUSECProvider extends Provider {
 
     private static final long serialVersionUID = -3240458633432287743L;
 
+    private static final Map<Object, Consumer<Object>> actions = new WeakHashMap<>();
+
+    @SuppressWarnings("unchecked")
+    public static <T> void doOnRestart(T object, Consumer<T> action) {
+        synchronized (actions) {
+            // This unchecked cast is safe because the action
+            // is only applied the supplied object.
+            actions.put(object, (Consumer<Object>) action);
+        }
```

Comment:
This should only be done if `InternalCRIUSupport.isCheckpointAllowed()`; otherwise `resetCRIUSEC()` will never be called.